includes/class-wc-gateway-ppec-checkout-handler.php

```diff
@@ -446,7 +446,7 @@ public function maybe_disable_other_gateways( $gateways ) {
 		}
 
 		// If the cart total is zero (e.g. because of a coupon), don't allow this gateway
-		if ( is_cart() || is_checkout() ) {
+		if ( is_cart() || ( is_checkout() && ! is_checkout_pay_page() ) ) {
```

Comment:
Looking at this check, I couldn't figure out why. Maybe we could add note/reference on the reasoning?